autonomy/cli/hash.py

```diff
@@ -55,6 +65,79 @@ def load_configuration(
     return cast(PackageConfiguration, configuration_obj)
 
 
+def update_hashes(  # pylint: disable=too-many-locals
+    packages_dir: Path,
+    no_wrap: bool = False,
+    vendor: Optional[str] = None,
+    config_loader: Callable[
+        [PackageType, Path], PackageConfiguration
+    ] = load_configuration,
+) -> int:
+    """Process all AEA packages, update fingerprint, and update hashes.csv files."""
+    return_code = 0
+    package_hashes: Dict[str, str] = {}
+
+    try:
+        public_id_to_hash_mappings: Dict = {}
+        dependency_tree = DependecyTree.generate(packages_dir)
+        packages = [
+            [package_id_and_path(package_id, packages_dir) for package_id in tree_level]
+            for tree_level in dependency_tree
+        ]
+        packages[0] = packages[0] + list(map(to_package_id, SCAFFOLD_PACKAGES))
+        for tree_level in packages:
+            for package_id, package_path in tree_level:
+                click.echo(
+                    "Processing package {} of type {}".format(
+                        package_path.name, package_id.package_type
+                    )
+                )
+
+                config_file = package_path / cast(
+                    str, PACKAGE_TYPE_TO_CONFIG_FILE.get(package_id.package_type.value)
+                )
+                item_config, extra_config = load_yaml(config_file)
+                if package_id.package_type == PackageType.SERVICE:
+                    agent_id = PackageId(
+                        PackageType.AGENT, PublicId.from_str(item_config["agent"])
+                    )
+                    item_config["agent"] = str(
+                        PublicId(
+                            author=agent_id.author,
+                            name=agent_id.name,
+                            version=agent_id.version,
+                            package_hash=public_id_to_hash_mappings[agent_id],
+                        )
+                    )
+                else:
+                    extend_public_ids(item_config, public_id_to_hash_mappings)
```

Comment:
This method is same as the one in `aea.cli.ipfs_hash`, Only difference is we check if the package type is service and update the agent package hash.